python/ray/serve/scripts.py

```diff
@@ -478,6 +487,24 @@ def build(import_path: str, app_dir: str, output_path: Optional[str]):
         ).dict(exclude_defaults=True)
     )
 
+    if kubernetes_format:
+        print(
+            "NOTE: Kubernetes requires the following fields to be "
```

Comment:
I think this warning can be worded more concisely and clearly. Does anyone have suggestions?

Reply:
+1, it would be great to remove this manual action from the customer side.

Reply:
We could json serialize it. The output would look like this:

```yaml
importPath: modded_fruit:deployment_graph

runtimeEnv: '{}'

host: 0.0.0.0

port: 8000

deployments:

- name: MangoStand
  userConfig: '{"price": 3, "fake_prop": "what_a_str!", "5": 2, "22": "hello"}'
  rayActorOptions:
    numCpus: 0.1

- name: OrangeStand
  userConfig: '{"price": 2, "fake_prop": "what_a_str!", "5": 2, "22": "hello"}'
  rayActorOptions:
    numCpus: 0.1

- name: PearStand
  userConfig: '{"price": 4, "fake_prop": "what_a_str!", "5": 2, "22": "hello"}'
  rayActorOptions:
    numCpus: 0.1

- name: FruitMarket
  numReplicas: 2
  rayActorOptions:
    numCpus: 0.1

- name: DAGDriver
  routePrefix: /
```

This is a bit more convoluted than e.g. [this YAML spec](https://github.com/ray-project/kuberay/blob/2793b2c3ec3f002c7c3399e0a7101fc3f5cbfc65/ray-operator/config/samples/ray_v1alpha1_rayservice.yaml#L13-L43) where each key in the spec is listed on a newline with a pipe (`|`) character:

> ```yaml
> - name: OrangeStand
>         numReplicas: 1
>         userConfig: |
>           price: 2
>         rayActorOptions:
>           numCpus: 0.1
> ```

**Note:** I anticipated that the JSON-serialized version would look worse than it is. Now that I've tried it, I actually don't think it looks too bad. I'd prefer to JSON-serialize and remove the manual step. What do you think @simon-mo @sihanwang41?

Reply:
I've updated the code to JSON-serialize the fields that Kubernetes config expects to be strings:

* `runtime_env`
* `ray_actor_options` > `runtime_env`
* `ray_actor_options` > `resources`